File: platform/server/internal/pipeline/status.go
package pipeline

type Status string

const (
	StatusPending  Status = "pending"
	StatusRunning  Status = "running"
	StatusSuccess  Status = "success"
	StatusFailed   Status = "failed"
	StatusSkipped  Status = "skipped"
	StatusCanceled Status = "canceled"
	StatusTimeout  Status = "timeout"
)

type OwnerType string

const (
	OwnerBuild  OwnerType = "build"
	OwnerDeploy OwnerType = "deploy"
)

File: web/src/status.ts
export const statusClass: Record<string, string> = {
  success: "status success",
  failed: "status failed",
  timeout: "status failed",
  canceled: "status muted",
  running: "status running",
  pending: "status pending",
  waiting_confirmation: "status running",
  skipped: "status muted"
};

export function classForStatus(status: string): string {
  return statusClass[status] ?? "status pending";
}

